massa-protocol-worker/src/protocol_worker.rs

```diff
@@ -226,27 +226,30 @@ mod nodeinfo {
             self.known_operations.contains(op)
         }
 
-        /// insert a block in wanted list of a node.
-        /// Note that it also insert the block as a not known block for that node.
-        pub fn insert_wanted_blocks(
+        /// insert a block in the wanted list of a node.
+        /// Also lists the block as not known by the node
+        pub fn insert_wanted_block(
             &mut self,
             block_id: BlockId,
             max_node_wanted_blocks_size: usize,
+            max_node_known_blocks_size: usize,
         ) {
-            self.wanted_blocks.insert(block_id, Instant::now());
-            while self.known_blocks.len() > max_node_wanted_blocks_size {
+            let now = Instant::now();
+            self.wanted_blocks.insert(block_id, now);
+            while self.wanted_blocks.len() > max_node_wanted_blocks_size {
                 // remove oldest item
                 let (&h, _) = self
-                    .known_blocks
+                    .wanted_blocks
                     .iter()
                     .min_by_key(|(h, t)| (*t, *h))
                     .unwrap(); // never None because is the collection is empty, while loop isn't executed.
-                self.known_blocks.remove(&h);
+                self.wanted_blocks.remove(&h);
```

Comment:
Maybe put that in a remove exceeding blocks method for clarity (and maybe not only here but also when ever this pattern is used)

Reply:
done

Reply:
I don't see the change you're talking about

Reply:
look at the absolute code. Since the code extracted by github for the preview didn't change and just moved to a new function, the change is not visible in the preview